docs/modules/data-structures/pages/listening-for-map-entries.adoc

```diff
@@ -133,45 +139,30 @@ func main() {
 	// error handling is omitted for brevity
 	ctx := context.TODO()
 	client, _ := hazelcast.StartNewClient(ctx)
-	entryListenerConfig := hazelcast.MapEntryListenerConfig{
-		IncludeValue: true,
-		Predicate:    predicate.Equal("surname", "Smith"),
-	}
-	m, _ := client.GetMap(ctx, "somemap")
-	entryListenerConfig.NotifyEntryAdded(true)
-	entryListenerConfig.NotifyEntryRemoved(true)
-	entryListenerConfig.NotifyEntryUpdated(true)
-	entryListenerConfig.NotifyEntryEvicted(true)
-	entryListenerConfig.NotifyEntryLoaded(true)
-
-	m.AddEntryListener(ctx, entryListenerConfig, func(event *hazelcast.EntryNotified) {
-		switch event.EventType {
-		case hazelcast.EntryAdded:
-			fmt.Println("Entry Added:", event)
-		case hazelcast.EntryRemoved:
-			fmt.Println("Entry Removed:", event)
-		case hazelcast.EntryUpdated:
-			fmt.Println("Entry Updated:", event)
-		case hazelcast.EntryEvicted:
-			fmt.Println("Entry Remove:", event)
-		case hazelcast.EntryLoaded:
-			fmt.Println("Entry Loaded:", event)
-		}
-	})
-
-
+	m, _ := client.GetMap(ctx, "map")
+	subscriptionID, _ := m.AddListenerWithPredicate(ctx, hazelcast.MapListener{
+		EntryAdded: func(event *hazelcast.EntryNotified) {
+			fmt.Println("Entry Added:", event.Key)
```

Comment:
Java code prints the event itself here, can we achieve the same somehow? 

Can we print an output like:
```
entryAdded:EntryEvent {Address[192.168.178.10]:5702} key=1,oldValue=null,
value=Person{name= smith }, event=ADDED, by Member [192.168.178.10]:5702
```

if it's not possible easily let's skip.

Reply:
Directly printing the event gives a little different output in Go.
```
Entry Added: &{<nil> 10 {
                "name":"Jack",
                "surname": "Smith"
        } <nil> map 127.0.0.1:5701:56df295c-370a-42b2-ab77-7d27812c7875 1 1}
```
So , we have to write a format function to get the same output as Java. Maybe we can leave it as above instead of `event.Key`.

Reply:
@mehmettokgoz I think that's a good idea, can you implement the Stringer interface for EntryNotified type? We can try to make it similar to Java's output that Serkan shared

Reply:
Implemented `Stringer()` interface that reflects the Java client output, see https://github.com/hazelcast/hazelcast-go-client/pull/864 for more details.

Reply:
We don't aim to have the same output with the reference implementation in these cases. In case we cannot skip the output in this case and the output have to look like reference implementation's, we can write a small utility function.